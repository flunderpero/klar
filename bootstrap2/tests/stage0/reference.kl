struct Address:
    city str
end

struct Person:
    name str
    address &Address
end

fn main():
    let address = Address("London")
    let person = Person("John", &address)
    print(f"{person.name} lives in {person.address.city}")
    let ref = &address
    print(f"Address is {ref.city}")
end
