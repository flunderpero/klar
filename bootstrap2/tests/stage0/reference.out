John lives in London
Address is London
